includes/Modules/Search_Console.php

```diff
@@ -102,15 +102,16 @@ function( array $token_response ) {
 		);
 
 		// Ensure that the data available state is reset when the property changes.
-		add_action(
-			'update_option_googlesitekit_search-console_settings',
+		$this->get_settings()->on_change(
 			function( $old_value, $new_value ) {
-				if ( $old_value['propertyID'] !== $new_value['propertyID'] ) {
+				if (
+					is_array( $old_value ) &&
+					is_array( $new_value ) &&
+					isset( array_diff_assoc( $new_value, $old_value )['propertyID'] )
```

Comment:
The conditions here were changed because the previous logic [caused a bunch of test cases to fail](https://github.com/google/site-kit-wp/actions/runs/7185872043/job/19570039354). See more discussion about this [here](https://github.com/google/site-kit-wp/issues/6517#issuecomment-1855392347).

It makes more sense to update the failing test cases by passing a `propertyID` to the option as that is the way SK expects it to work, but there are some generic test utilities that cause failures as well. Please feel free to let me know if you think the utilities should be updated as well.